### DISCUSSION TITLE
[v2] There was a problem with creating a new Window on the front end

### DISCUSSION THREAD

**Discussion by nongyehong** [tauri-apps/tauri#9381] [Q&A] [answered]
```vue
<script setup lang="ts">
// This starter template is using Vue 3 <script setup> SFCs
// Check out https://vuejs.org/api/sfc-script-setup.html#script-setup
import Greet from "./components/Greet.vue";
import { Window } from '@tauri-apps/api/window';

const createWin = () => {
  return new Window('home');
}
</script>

<template>
  <div data-tauri-drag-region class="container">
    <h1>Welcome to Tauri!</h1>
    <div class="row">
      <a href="https://vitejs.dev" target="_blank">
        <img src="/vite.svg" class="logo vite" alt="Vite logo" />
      </a>
      <a href="https://tauri.app" target="_blank">
        <img src="/tauri.svg" class="logo tauri" alt="Tauri logo" />
      </a>
      <a href="https://vuejs.org/" target="_blank">
        <img src="./assets/vue.svg" class="logo vue" alt="Vue logo" />
      </a>
    </div>
    <button @click="createWin">create window</button>

    <Greet />
  </div>
</template>

<style scoped>
.logo.vite:hover {
  filter: drop-shadow(0 0 2em #747bff);
}

.logo.vue:hover {
  filter: drop-shadow(0 0 2em #249b73);
}
</style>

```

When I click Create, the console gives me the following error:

![image](https://github.com/tauri-apps/tauri/assets/87641407/3e4572a0-2249-4830-a2ff-a86a5e565c1a)

**Comment by nongyehong** [comment]
tauri info

[✔] Environment
    - OS: Windows 10.0.26080 X64
    ✔ WebView2: 123.0.2420.65
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.77.1 (7cf61ebde 2024-03-27)
    ✔ cargo: 1.77.1 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.10.0
    - pnpm: 8.15.4
    - yarn: 1.22.21
    - npm: 10.2.3

[-] Packages
    - tauri [RUST]: 2.0.0-beta.14
    - tauri-build [RUST]: 2.0.0-beta.11
    - wry [RUST]: 0.39.0
    - tao [RUST]: 0.27.0
    - @tauri-apps/api [NPM]: 2.0.0-beta.7
    - @tauri-apps/cli [NPM]: 2.0.0-beta.12

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1600/
    - framework: Vue.js
    - bundler: Vite

**Comment by FabianLars** [comment, marked as answer]
Did you try using WebviewWindow instead of Window? If that also doesn't work, please share your capability file.

**Reply by nongyehong** [reply]
It's a coincidence that I've tried it all.

--capabilities.json--

```json
{"default":{"identifier":"default","description":"Capability for the main window","local":true,"windows":["main","home"],"permissions":["path:default","event:default","window:default","app:default","image:default","resources:default","menu:default","tray:default","shell:allow-open","window:allow-create","window:allow-start-dragging"]}}
```
Whether it is Window or WebviewWindow or according to the documentation, this error is reported. It bothers me for a long time, I am very confused

**Reply by FabianLars** [reply]
Hmm, try adding this to the permissions array
```
"webview:default", "webview:allow-create-webview-window"
```

**Reply by nongyehong** [reply]
I immediately tried to add these: "webview:allow-create-webview-window",
"webview:allow-create-webview",
"webview:default", but still get this error

**Reply by nongyehong** [reply]
Maybe this is not my problem, because this is a newly created vue+tauri-V2 project

**Reply by FabianLars** [reply]
after adding `"webview:default", "webview:allow-create-webview-window"` the `new WebviewWindow()` call works for me. I'm missing something obvious here 🤔

**Reply by nongyehong** [reply]
Thank you, is when I create the window to give some do not give permission to the function, resulting in the creation of the window when the error occurs

**Reply by nongyehong** [reply]
@FabianLars When creating a new window I need to set fileDropEnabled: false What permission should I give it in capabilies.json, I need to drag and drop elements in the page
